text/0070-pd-record-historical-hot-region.md

```diff
@@ -0,0 +1,163 @@
+
+## Summary
+
+Create a new table `TIDB_HOT_REGIONS_HISTORY` in the `INFORMATION_SCHEMAT` database to retrieve history hotspot information stored by PD periodically.  
+
+## Motivation
+
+TiDB has a memory table `TIDB_HOT_REGIONS` that provides information about hotspot regions. 
+But it only shows the current hotspot information. This leads to the fact that when DBAs want to query historical hotspot details, 
+they have no way to find the corresponding hotspot information.
+
+## Background
+According to the [documentation](https://docs.pingcap.com/tidb/stable/information-schema-tidb-hot-regions) for the current `TIDB_HOT_REGIONS` table, 
+the table provides information about hotspot regions shown in the following table: 
+
+```SQL
++----------------+-------------+------+------+---------+-------+
+| Field          | Type        | Null | Key  | Default | Extra |
++----------------+-------------+------+------+---------+-------+
+| TABLE_ID       | bigint(21)  | YES  |      | NULL    |       |
+| INDEX_ID       | bigint(21)  | YES  |      | NULL    |       |
+| DB_NAME        | varchar(64) | YES  |      | NULL    |       |
+| TABLE_NAME     | varchar(64) | YES  |      | NULL    |       |
+| INDEX_NAME     | varchar(64) | YES  |      | NULL    |       |
+| REGION_ID      | bigint(21)  | YES  |      | NULL    |       |
+| TYPE           | varchar(64) | YES  |      | NULL    |       |
+| MAX_HOT_DEGREE | bigint(21)  | YES  |      | NULL    |       |
+| REGION_COUNT   | bigint(21)  | YES  |      | NULL    |       |
+| FLOW_BYTES     | bigint(21)  | YES  |      | NULL    |       |
++----------------+-------------+------+------+---------+-------+
+10 rows in set (0.00 sec)
+```
+There are two types of hotspot regions: read and write. 
+The memory table retriever processes the following steps to fetch current hotspot regions from PD server:
+
+1. TiDB send an HTTP request to the PD to obtain the hotspot regions information of the current cluster. 
+    `pd/api/v1/hotspot/regions/read` for read hotspot and `pd/api/v1/hotspot/regions/write` for write hotspot. 
+    PD returns the following fields：
+    ```go
+      // HotPeerStatShow records the hot region statistics for output
+    type HotPeerStatShow struct {
+      StoreID        uint64    `json:"store_id"`
+      RegionID       uint64    `json:"region_id"`
+      HotDegree      int       `json:"hot_degree"`
+      ByteRate       float64   `json:"flow_bytes"`
+      KeyRate        float64   `json:"flow_keys"`
+      QueryRate      float64   `json:"flow_query"`
+      AntiCount      int       `json:"anti_count"`
+      LastUpdateTime time.Time `json:"last_update_time"`
+    }
+    ```
+1. Parse response and fetch the start key and end key of the hot region from region cache or PD by `REGION_ID` to calculate the region's corresponding schema information like：`TABLE_ID`, `TABLE_NAME`, `DB_NAME`, `INDEX_ID`, `INDEX_NAME`.
+1. Return the hotspot information to the upper call.
```

Comment:
These should be moved to detailed design, and I think it is enough to mention that the hot regions data is fetched from PD circlely instead expose the detailed api design here.

Reply:
I also suggestion to provide common examples for querying `HotPeerStatShow` in daily investigament.

Reply:
Move `TIDB_HOT_REGIONS` workflow to Detailed design section and remove the api.

Reply:
It seems to be a more commom way to query `HotPeerStatShow` by `pd-ctl`. Does `TIDB_HOT_REGIONS_HISTORY` need to support `pd-ctl` in the future, if so, the data will take some time to transmit.